pwiz_tools/Skyline/Model/FullScanProperties.cs

```diff
@@ -0,0 +1,141 @@
+                        precursor.IsolationWindowUpper.Value.ToString(Formats.Mz));
+            }
+
+            res.RetentionTime = spectrum.RetentionTime.HasValue ? spectrum.RetentionTime.Value.ToString(Formats.RETENTION_TIME) : null;
+
+            res.MSStage = spectrum.Level.ToString();
+            res.ScanId = spectrum.Id;
+
+            if (spectrum.InstrumentInfo != null)
+            {
+                res.Instrument = new InstrumentInfo();
+                res.Instrument.InstrumentManufacturer = spectrum.InstrumentVendor;
+                res.Instrument.InstrumentModel = spectrum.InstrumentInfo.Model;
+                if (new[]
+                    {
+                        spectrum.InstrumentInfo.Ionization, 
+                        spectrum.InstrumentInfo.Analyzer,
+                        spectrum.InstrumentInfo.Detector
+                    }.Any( s => !string.IsNullOrEmpty(s)))
+                {
+                    res.Instrument.InstrumentComponents = new InstrumentInfo.InstrumentComponentsInfo();
+                    res.Instrument.InstrumentComponents.Ionization = spectrum.InstrumentInfo.Ionization;
+                    res.Instrument.InstrumentComponents.Analyzer = spectrum.InstrumentInfo.Analyzer;
+                    res.Instrument.InstrumentComponents.Detector = spectrum.InstrumentInfo.Detector;
+                }
+            }
+
+            res.Instrument.InstrumentSerialNumber = spectrum.InstrumentSerialNumber;
+            res.IsCentroided = spectrum.Centroided.ToString(CultureInfo.CurrentCulture);
+            return res;
+        }
+        [Category("FileInfo")] public string FileName { get; set; }
+        // need to exclude the file path from test assertions because it is machine-dependent
+        [UseToCompare(false)] [Category("FileInfo")] public string FilePath { get; set; }
+        [Category("FileInfo")] public string ReplicateName { get; set; }
+        [Category("PrecursorInfo")] public string PrecursorMz { get; set; }
+        [Category("PrecursorInfo")] public string Charge { get; set; }
+        [Category("PrecursorInfo")] public string Label { get; set; }
+        [Category("PrecursorInfo")] public string RetentionTime { get; set; }
+        [Category("PrecursorInfo")] public string CCS { get; set; }
+        [Category("PrecursorInfo")] public string IonMobility { get; set; }
+        [Category("PrecursorInfo")] public string IsolationWindow { get; set; }
+        [Category("AcquisitionInfo")] public string IonMobilityRange { get; set; }
+        [Category("AcquisitionInfo")] public string IonMobilityFilterRange { get; set; }
+        [Category("PrecursorInfo")] public string HighEnergyOffset { get; set; }
+        [Category("AcquisitionInfo")] public string ScanId { get; set; }
+        [Category("AcquisitionInfo")] public string CE { get; set; }
+        [Category("AcquisitionInfo")] public string MSStage { get; set; }
```

Comment:
Everywhere else in Skyline, (e.g. "Settings > Peptide Settings > Quantification") I think we call this "MS Level" instead of "MS Stage".

Reply:
Yes, I think MS Stage came from the SeeMS since it provided initial inspiration for this.